src/main/java/org/datadog/jmxfetch/Configuration.java

```diff
@@ -33,4 +42,212 @@ public String toString() {
+
+                    if (parameterValues.size() != 1 || (commonValue != null && !commonValue.equals(parameterValues.get(0)))) {
+                        hasCommonValue = false;
+                        continue;
+                    }
+
+                    commonValue = parameterValues.get(0);
+
+                }
+                if (hasCommonValue) {
+                    commonScope.put(parameter, commonValue);
+                }
+           }
+           commonScopeByDomain.put(domainName, commonScope);
+       }
+       return commonScopeByDomain;
+    }
+
+    /**
+     * Stringify a bean pattern.
+     *
+     * @param domain                domain name
+     * @param beanScope             map of bean keys-> values
+     *
+     * @return                      string pattern identifying the bean scope
+     */
+    private static String beanScopeToString(String domain, LinkedHashMap<String, String> beanScope){
+        String result = "";
+
+        // Domain
+        domain = (domain != null) ? domain : "*";
+        result += domain + ":";
+
+        // Scope parameters
+        Iterator<Entry<String, String>> beanScopeItr = beanScope.entrySet().iterator();
+        while(beanScopeItr.hasNext()){
+            Entry<String, String> beanScopeEntry = beanScopeItr.next();
+            String param = (String) beanScopeEntry.getKey();
+            String value = (String) beanScopeEntry.getValue();
+
+            result += param + "=" + value + ",";
+        }
+        result += "*";
+
+        return result;
+    }
+
+    /**
+     * Find, among the configuration list, a potential common bean pattern by domain name.
+     *
+     * @param configurationList         the configuration list to process
+     *
+     * @return                          common bean pattern strings
+     */
+    public static LinkedList<String> getGreatestCommonScopes(LinkedList<Configuration> configurationList){
+        LinkedList<Configuration> includeConfigList = getIncludeConfigurationList(configurationList);
+        HashMap<String, LinkedList<Filter>> includeFiltersByDomain = getIncludeFiltersByDomain(includeConfigList);
+        HashMap<String, Set<String>> parametersIntersectionByDomain = getCommonBeanKeysByDomain(includeFiltersByDomain);
+        HashMap<String, LinkedHashMap<String, String>> commonBeanScopeByDomain = getCommonScopeByDomain(parametersIntersectionByDomain, includeFiltersByDomain);
+
+        LinkedList<String> result = new LinkedList<String>();
+        Iterator<Entry<String, LinkedHashMap<String, String>>> commonBeanScopeItr = commonBeanScopeByDomain.entrySet().iterator();
+        while(commonBeanScopeItr.hasNext()){
+            Entry<String, LinkedHashMap<String, String>> beanScopeEntry = commonBeanScopeItr.next();
+            String domain = (String) beanScopeEntry.getKey();
+            LinkedHashMap<String, String> beanScope = (LinkedHashMap<String, String>) beanScopeEntry.getValue();
```

Comment:
There's no need to cast the values returned by `getKey()` and `getValue()` since they're already of type `String` and `LinkedHashMap<String, String>` respectively.

Also applies to lines 131-132, 166-167, 217-218.